models.go

```diff
@@ -396,3 +397,30 @@ type CredentialRepresentation struct {
 	Type              string             `json:"type,omitempty"`
 	Value             string             `json:"value,omitempty"`
 }
+
+// GetEventsParams represents the optional parameters for getting events
+type GetEventsParams struct {
+	Type      string `json:"Type,omitempty"`
+	Client    string `json:"client,omitempty"`
+	User      string `json:"user,omitempty"`
+	DateFrom  string `json:"dateFrom,omitempty"`
+	DateTo    string `json:"dateTo,omitempty"`
+	IPAddress string `json:"ipAddress,omitempty"`
+	First     int    `json:"first,string,omitempty"`
+	Max       int    `json:"max,string,omitempty"`
+	Realm     string `json:"realm"`
```

Comment:
add `omitempty` otherwise the query params will always have "realm" key: https://play.golang.org/p/q0NFca9eOwI